dapps/marketplace/src/pages/listing/_BuyFractional.js

```diff
@@ -55,7 +63,15 @@ const Fractional = ({ listing, from, range, availability, refetch }) => {
         return (
           <div className="listing-buy fractional">
             <div className="price">
-              <Price listing={listing} descriptor />
```

Comment:
Instead of wrapping all these places, could we put the logic in the `OgnBadge` component and just return null if there is no reward for the listing?

Reply:
nice idea thanks 👍